api/pkg/apis/v1alpha1/providers/target/k8s/k8s.go

```diff
@@ -592,6 +697,18 @@ func (i *K8sTargetProvider) upsertService(ctx context.Context, namespace string,
 	if err != nil {
 		return err
 	}
+	waitErr := i.pollCheck(ctx, false, func() error {
+		_, err := i.Client.CoreV1().Services(namespace).Get(ctx, name, metav1.GetOptions{})
```

Comment:
For service, we need to polling the service creation either, helm tool used to ensure the service is ready like this: https://github.com/helm/helm/blob/d790f7d843182f1c126cfa64989ffdd4e9583747/pkg/kube/wait.go#L143C1-L162C2

Reply:
fixed. follow the same check pattern with helm